presto-pinot/src/main/java/com/facebook/presto/pinot/PinotClusterInfoFetcher.java

```diff
@@ -0,0 +1,199 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.pinot;
+
+import com.alibaba.fastjson.JSONArray;
+import com.alibaba.fastjson.JSONObject;
+import com.facebook.presto.spi.PrestoException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.inject.Inject;
+import io.airlift.log.Logger;
+import org.apache.http.HttpEntity;
+import org.apache.http.HttpHeaders;
+import org.apache.http.HttpResponse;
+import org.apache.http.client.ClientProtocolException;
+import org.apache.http.client.ResponseHandler;
+import org.apache.http.client.methods.HttpUriRequest;
+import org.apache.http.client.methods.RequestBuilder;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClients;
+import org.apache.http.util.EntityUtils;
+import org.apache.pinot.client.DynamicBrokerSelector;
+import org.apache.pinot.common.data.Schema;
+import org.apache.pinot.common.utils.NetUtil;
+
+import java.io.IOException;
+import java.net.SocketException;
+import java.net.UnknownHostException;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Random;
+
+import static com.facebook.presto.pinot.PinotErrorCode.PINOT_FAILURE_GETTING_TABLE;
+import static org.apache.pinot.common.config.TableNameBuilder.extractRawTableName;
+
+/**
+ * This class handles fetching necessary information from the Pinot cluster, including table schema, broker host, routing table, etc
+ */
+public class PinotClusterInfoFetcher
+{
+    private static final String APPLICATION_JSON = "application/json";
+
+    private static final String GET_ALL_TABLES_API_TEMPLATE = "http://%s/tables";
+    private static final String TABLE_SCHEMA_API_TEMPLATE = "http://%s/tables/%s/schema";
+    private static final String ROUTING_TABLE_API_TEMPLATE = "http://%s/debug/routingTable/%s";
+    private static final String TIME_BOUNDARY_API_TEMPLATE = "http://%s/debug/timeBoundary/%s";
+
+    private static final Logger log = Logger.get(PinotClusterInfoFetcher.class);
+    private static final CloseableHttpClient httpClient = HttpClients.createDefault();
```

Comment:
put an empty line after. We usually separate constant and class variables. Also, let's capitalize `HTTP_CLIENT`